lib/ecto/repo/schema.ex

```diff
@@ -169,7 +169,11 @@ defmodule Ecto.Repo.Schema do
     schema = struct.__struct__
     fields = schema.__schema__(:fields)
     assocs = schema.__schema__(:associations)
-    return = schema.__schema__(:read_after_writes)
+    return = case opts[:returning] do
+      true                    -> schema.__schema__(:fields)
+      list when is_list(list) -> list
+      _                       -> schema.__schema__(:read_after_writes)
```

Comment:
I think we should always concatenate the two lists and then call Enum.uniq on it.